boards/FSM/Src/Io/Io_FlowMeter.c

```diff
@@ -1,20 +1,91 @@
 #include <assert.h>
 #include "main.h"
-#include "Io_SharedPwmInput.h"
+#include "Io_SharedFrequencyOnlyPwmInput.h"
 #include "Io_FlowMeter.h"
 
-static struct PwmInput *  primary_flow_meter_pwm_input;
-static TIM_HandleTypeDef *primary_flow_meter_htim;
+#define PRIMARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_1
+#define SECONDARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_2
+
+extern TIM_HandleTypeDef htim4;
+
+static struct FrequencyOnlyPwmInput *primary_flow_meter_pwm_input,
+    *secondary_flow_meter_pwm_input;
+static TIM_HandleTypeDef *flow_meters_timer_handle;
+
+static size_t primary_channel_period_elapsed_tick   = 0U;
+static size_t secondary_channel_period_elapsed_tick = 0U;
+
+void Io_FlowMeter_Init(void)
+{
+    // Configuring timer handle for both flow meters
+    flow_meters_timer_handle = &htim4;
+
+    primary_flow_meter_pwm_input = Io_SharedFrequencyOnlyPwmInput_Create(
+        &htim4, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_1);
+    secondary_flow_meter_pwm_input = Io_SharedFrequencyOnlyPwmInput_Create(
+        &htim4, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_2);
+}
 
 float Io_FlowMeter_GetPrimaryFlowRate(void)
 {
-    return Io_SharedPwmInput_GetFrequency(primary_flow_meter_pwm_input) / 7.5f;
+    return Io_SharedFrequencyOnlyPwmInput_GetFrequency(
+               primary_flow_meter_pwm_input) /
+           7.5f;
+}
+
+float Io_FlowMeter_GetSecondaryFlowRate(void)
+{
+    return Io_SharedFrequencyOnlyPwmInput_GetFrequency(
+               secondary_flow_meter_pwm_input) /
+           7.5f;
+}
+
+void Io_FlowMeter_Primary_FlowMeter_Tick(void)
+{
+    static uint32_t primary_previous_rising_edge, primary_current_rising_edge;
+    static bool     primary_first_tick_state = true;
+    primary_channel_period_elapsed_tick      = 0U;
+
+    Io_SharedFrequencyOnlyPwmInput_Tick(
+        primary_flow_meter_pwm_input, &primary_first_tick_state,
+        &primary_current_rising_edge, &primary_previous_rising_edge);
+}
+
+void Io_FlowMeter_Secondary_FlowMeter_Tick(void)
+{
+    static uint32_t secondary_previous_rising_edge,
+        secondary_current_rising_edge;
+    static bool secondary_first_tick_state = true;
+    secondary_channel_period_elapsed_tick  = 0U;
+
+    Io_SharedFrequencyOnlyPwmInput_Tick(
+        secondary_flow_meter_pwm_input, &secondary_first_tick_state,
+        &secondary_current_rising_edge, &secondary_previous_rising_edge);
 }
 
 void Io_FlowMeter_InputCaptureCallback(TIM_HandleTypeDef *htim)
```

Comment:
I think it would be better if this was broken down into two functions, like how we have a `Tick` function for each flow meter.

```
Io_FlowMeter_PimaryInputCaptureCallback()
Io_FlowMeter_SecondaryInputCaptureCallback()
```

It makes the intent more clear